app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -189,108 +179,16 @@ public void onSuccess(AuthResult result) {
                 .addOnCompleteListener(new SignInResultNotifier(this));
     }
 
-    public boolean isSignedIn() {
+    private boolean isSignedIn() {
         return mAuth.getCurrentUser() != null;
     }
 
-    public void updateUI() {
+    private void updateUI() {
         // Sending only allowed when signed in
         mSendButton.setEnabled(isSignedIn());
         mMessageEdit.setEnabled(isSignedIn());
     }
 
-    public static class Chat {
```

Comment:
Putting these in separate files makes the `ChatActivity` easier to understand since it doesn't contain the java bean or holder.